### ISSUE TITLE
cant spawn or attach any APP.

### ISSUE THREAD

**Issue by Shubhamsonani3** [nowsecure/r2frida#527]
Hi team, I am getting the error - unable to connect to remote frida-server. I am using latest version of frida-server and frida. I have also tried using lower versions of all. Please help in solving this. I am trying in the windows. 
![image](https://github.com/nowsecure/r2frida/assets/109787829/ebb93f4a-cf81-4bec-ac5c-47c6ebee45a1)

1. C:\r2frida>frida --version
-> 16.1.5

2. C:\r2frida>radare2 -v
-> radare2 5.8.8 1 @ windows-x86-64

3. C:\r2frida>r2 -v
-> radare2 5.8.8 1 @ windows-x86-64

C:\r2frida> r2 frida://attach/usb//HelloWorldJNI
ERROR: Cannot attach: Unable to connect to remote frida-server: closed
ERROR: Cannot open 'frida://attach/usb//HelloWorldJNI'

**Comment by ditorri1** [comment]
Are you using a USB A connector? Or USB C?

**Comment by Shubhamsonani3** [comment]
I am using android studio's emulator, Android 7 and 10

**Comment by trufae** [comment]
Is the frida server running inside the target device? If usb doesnt work try the tcp socket. But afaik latest r2frida is broken on windows because of s bug in frida compiling the agent

**Comment by Shubhamsonani3** [comment]
Yes frida 16.1.5 is running on 0.0.0.0:12345. I tried with r2frida 5.7.2 but it shows same.

**Comment by Shubhamsonani3** [comment]
According to your steps I managed to cloned your repository. The second step was to type "make" command to build r2frida. Now, in windows it is not working totally (i have tried installing with visual studio 2019). As described before, r2frida doesn't spawn or attach the app. So, I tried with Windows-subsystem-Linux and installed kali Linux. I used this blog(**https://jaro.blog/blog/adb-in-wsl2.html**) to use adb from kali Linux wsl. I cloned your git repository, and it was throwing some code Error while building the tool. To fix the code, go to "src->io_frida.c", change this line of code to which I have provided. -> **R_LOG_DEBUG ("write %llx @ 0x%08"PFMT64x, (unsigned long long)io->off, (unsigned long long)count);** second is to remove **R_LOG_DEBUG("")**. Then try to build the tool with mentioned steps. Actually there was some type error in logging which instead of logging was throwing error. I hope this works, for me it worked so.
